src/centos/macros.tpl

```diff
@@ -41,6 +41,15 @@ RUN \
 {% else %}
     virtualenv-$PYTHON_VERSION ${APP_ROOT} && \
 {% endif %}
+    # We have to upgrade pip to a newer verison because: \
+    # * pip < 9 does not support different packages' versions for Python 2/3 \
+    # * pip < 19.3 does not support manylinux2014 wheels. Only manylinux2014 wheels \
+    #   support platforms like ppc64le, aarch64 or armv7 \
+    # We are newly using wheel from the latest stable Fedora release (RPM python-pip-wheel) \
+    # because it's tested better that whatever version from PyPI and contains useful patches. \
+    # We have to do it here so the permissions are correctly fixed and pip is able  to reinstall \
+    # itself in the next build phases and in the assemble script if user wants the latest version \
+    ${APP_ROOT}/bin/pip install /opt/wheels/pip-* && rm -r /opt/wheels && \
```

Comment:
> rm -r /opt/wheels 

Don't we need `rm -rf` for a directory?

Reply:
we need only `-r` eg `--recursive` to remove directories and their contents recursively